src/dimension_slice.h

```diff
@@ -53,8 +53,7 @@ extern DimensionVec *ts_dimension_slice_collision_scan_limit(int32 dimension_id,
 extern bool ts_dimension_slice_scan_for_existing(DimensionSlice *slice);
 extern DimensionSlice *ts_dimension_slice_scan_by_id_and_lock(int32 dimension_slice_id,
 															  ScanTupLock *tuplock,
-															  MemoryContext mctx)
-	TS_ATTRIBUTE_NONNULL((2));
```

Comment:
Can you also remove the definition of this macro? It's not used anywhere anymore.